pylintfileheader/file_header_checker.py

```diff
@@ -47,27 +56,40 @@ class FileHeaderChecker(BaseChecker):
         ),
     )
 
-    def process_module(self, node):
-        """Process the astroid node stream."""
-        if self.config.file_header:
-            content = None
-            with node.stream() as stream:
-                # Explicit decoding required by python 3
-                content = stream.read().decode('utf-8')
+    def __init__(self, linter=None):
+        # pylint: disable=super-with-arguments
+        super(FileHeaderChecker, self).__init__(linter=linter)
+        self.pattern = None
+        self.header = None
 
-            if self.config.file_header_ignore_empty_files and not content:
-                return
+    def open(self):
+        self.header = self.config.file_header
+        if not self.header and self.config.file_header_path:
+            with open(self.config.file_header_path, 'r') as header_file:
+                self.header = header_file.read()
 
+        if self.header:
             if sys.version_info[0] < 3:
-                pattern = re.compile(
-                    r'\A' + self.config.file_header, re.LOCALE | re.MULTILINE)
+                opts = re.LOCALE | re.MULTILINE
             else:
-                # The use of re.LOCALE is discouraged in python 3
-                pattern = re.compile(
-                    r'\A' + self.config.file_header, re.MULTILINE)
+                opts = re.MULTILINE
+            self.pattern = re.compile(r'\A' + self.header, opts)
+
+    def process_module(self, node):
+        """Process the astroid node stream."""
+
+        if self.pattern is None:
+            return
+
+        content = None
+        with node.stream() as stream:
+            # Explicit decoding required by python 3
+            content = stream.read().decode('utf-8')
+
+        if self.config.file_header_ignore_empty_files and not content:
```

Comment:
shouldn't this check be unnecessary, because after `open()` was called `self.pattern` is either set or not?

Reply:
no, if the `patter` in not `None` then we still need to ignore or not the empty files